openvdb/openvdb/tools/MeshToVolume.h

```diff
@@ -3116,10 +3139,168 @@ traceExteriorBoundaries(FloatTreeT& tree)
 } // void traceExteriorBoundaries()
 
 
+////////////////////////////////////////
+
+template <typename T, Index Log2Dim, typename InteriorTest>
+void
+floodFillLeafNode(tree::LeafNode<T,Log2Dim>& leafNode, const InteriorTest& interiorTest) {
+
+    enum VoxelState {
+        NOT_VISITED = 0,
+        POSITIVE = 1,
+        NEGATIVE = 2,
+        NOT_ASSIGNED = 3
+    };
+
+    const auto DIM = leafNode.DIM;
+    const auto SIZE = leafNode.SIZE;
+
+    std::vector<VoxelState> voxelState(SIZE, NOT_VISITED);
+
+    std::vector<std::pair<Index, VoxelState>> offsetStack;
+    offsetStack.reserve(SIZE);
+
+    for (Index offset=0; offset<SIZE; offset++) {
```

Comment:
I would have liked to see a short paragraph describing the high-level idea behind this new flood-filling algorithm

Reply:
Additional comments have been added!